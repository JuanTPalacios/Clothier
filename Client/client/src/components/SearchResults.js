import React from 'react'

export default function SearchResults({setSearchToggle}) {
  return (
    <div>
      Search Results here!
      <button className="button" type='click' onClick={()=>{setSearchToggle(false)}} > ‚ùå</button>
    </div>
  )
}

