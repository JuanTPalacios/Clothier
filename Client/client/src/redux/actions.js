const actions = {
  getItems : (items) => ({
    type: 'getItems',
    payload: items
  }),
};

export default actions;