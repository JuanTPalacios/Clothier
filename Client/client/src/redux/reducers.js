import { combineReducers } from 'redux';

const initialState = {
  items: [] ,
  setAuthenticated: false
}

const items = (state = initialState, action) => {
  switch (action.type) {
    case 'getItems':
    return {items: [...state.items, ...action.payload]};
  default:
    return state;
  }
};

// Combining both reducers
const reducers = combineReducers({
  items,
});

export default reducers;