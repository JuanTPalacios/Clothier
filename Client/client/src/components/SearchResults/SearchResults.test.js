import {fireEvent, render, screen} from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from '../../redux/reducers';
import SearchResults from './SearchResults';


describe('MyCloset component', () => {
  test('should match the snapshot', () => {
    let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__());
    const { container } = render(<Provider store={store}><SearchResults /></Provider>);
    expect(container.firstChild).toMatchSnapshot();
  })

  test('it should correctly render the headers', () => {
    let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__());
   render(<Provider store={store}><SearchResults /></Provider>);
   screen.getByText(/People:/)
   screen.getByText(/Items:/)
 })

})
  