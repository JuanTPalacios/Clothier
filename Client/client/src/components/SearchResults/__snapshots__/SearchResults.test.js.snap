// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCloset component should match the snapshot 1`] = `
<div
  class="columns"
>
  <div
    class="column"
  >
    <h2>
      People:
    </h2>
    <div
      class="search-follow"
    />
  </div>
  <div
    class="column"
  >
    <h2>
      Items:
    </h2>
    <div
      class="search-follow"
    />
  </div>
</div>
`;
